docs/proposals/image-promotion.md

```diff
@@ -0,0 +1,201 @@
+
+## Use Cases
+
+### Automatic promotion
+
+As a developer of Rails application, when I update the source code and push it to a remote
+source code repository, I want to to have my changes tested before deploying them to production.
+I also want to have the Docker image that contains applications with my changes to be promoted
+based on the verification results.
+
+#### Example of using Jenkins
+
+As a user I have a Jenkins server running in my project by following the instructions from the [Jenkins example](https://github.com/openshift/origin/blob/master/examples/jenkins/README.md).
+Then I create the *frontend-production* DeploymentConfig in my *Application* project.
+
+This DeploymentConfig has the ImageChangeTrigger set to watch changes in the "prod-ready" ImageStreamTag:
+
+```json
+{
+  "type": "ImageChange",
+  "imageChangeParams": {
+    "automatic": true,
+    "from": {
+      "kind": "ImageStreamTag",
+      "name": "origin-ruby-example:prod-ready"
+    },
+    "containerNames": [
+      "frontend"
+    ]
+  }
+}
+```
+
+I also have a BuildConfig defined for my application image that does not specify any triggers.
+
+When a change is detected in remote source code repository, the Jenkins job will connect to OpenShift and
+start a new build.
+Based on the build status, Jenkins job then tag the updated application image as *prod-ready*.
+When a new image with this tag is available, the *frontend-production* manages to roll out a new deployment.
+
+#### Example of using OpenShift
+
+As a user I create two DeploymentConfig's, both as part of my "Application" project:
+
+* *frontend-test*
+* *frontend-production*
+
+The "frontend-test" DeploymentConfig has the ImageChangeTrigger set to watch changes
+in the *qa-ready* ImageStreamTag. This DeploymentConfig also defines the *post* lifecycle
+hook that executes a script in the container deployed and performs additional verification
+and tests.
+
+*Example of DeploymentConfig post-lifecycle hook*:
+```json
+"strategy": {
+  "type": "Rolling",
+  "rollingParams": {
+    "post": {
+      "failurePolicy": "Abort",
+      "execNewPod": {
+        "containerName": "test-container",
+        "command": [
+          "/opt/app-root/bin/verify-deployment"
+        ],
+      }
+    }
+  }
+}
+```
+
+Based on the result of the `verify-deployment` command, which runs in the container based on the
+application image, the command will promote the image by tagging it as *production-ready* (eg. using
+the `oc tag` command). The command might also notify me about the verification failure.
+
+The *frontend-production* DeploymentConfig has the ImageChangeTrigger set to watch changes
+in the *prod-ready* ImageStreamTag. When a new Docker image with this tag is available,
+it can automatically roll out a new production deployment. In case you don't want to roll out
+the *frontend-production* automatically, you can avoid specifying the ImageChangeTrigger and
+roll out the new deployment by hand.
```

Comment:
Using oc tag?  I would flush out this example so that you can show actual command lines.

Use the deployment images from `examples/deployment` if you have to - real examples win every time.